src/kafka/kafka_clients.clj

```diff
@@ -0,0 +1,149 @@
+                           "key.deserializer"   org.apache.kafka.common.serialization.StringDeserializer,
+                           "value.deserializer" org.apache.kafka.common.serialization.StringDeserializer,
+                           "client.id"          "m-w1kx3xw0nr",
+                           "bootstrap.servers"  "localhost:9092"}}]
+
+
+  (build-producer)
+  #_=> #_#object[org.apache.kafka.clients.producer.KafkaProducer
+                 0x53c5e34f
+                 "org.apache.kafka.clients.producer.KafkaProducer@53c5e34f"]
+
+  (build-consumer)
+  #_=> #_#object[org.apache.kafka.clients.consumer.KafkaConsumer
+                 0x186c605f
+                 "org.apache.kafka.clients.consumer.KafkaConsumer@186c605f"]
+
+  producer-atom
+  #_=> #_#object[clojure.lang.Atom
+                 0x2eba3b87
+                 {:status :ready,
+                  :val    #object[org.apache.kafka.clients.producer.KafkaProducer
+                                  0x56863721
+                                  "org.apache.kafka.clients.producer.KafkaProducer@56863721"]}]
+
+  consumer-atom
+  #_=> #_#object[clojure.lang.Atom
+                 0x67822252
+                 {:status :ready,
+                  :val    #object[org.apache.kafka.clients.consumer.KafkaConsumer
+                                  0x1a8ffcb1
+                                  "org.apache.kafka.clients.consumer.KafkaConsumer@1a8ffcb1"]}]
+
+  ;; generate record
+  (do (def record (new ProducerRecord "demo" "Hello world!"))
+      record)
+  #_=> #_#object[org.apache.kafka.clients.producer.ProducerRecord
+                 0x3fb54ba6
+                 "ProducerRecord(topic=demo, partition=null, headers=RecordHeaders(headers = [], isReadOnly = false), key=null, value=Hello world!, timestamp=null)"]
+
+  ;; producer send record
+  (send-msg! record)
+  #_=> #_#object[org.apache.kafka.clients.producer.internals.FutureRecordMetadata
+                 0xde5318c
+                 "org.apache.kafka.clients.producer.internals.FutureRecordMetadata@de5318c"]
+
+  ;; consumer receive message
+  (read-msg!)
+  #_=> ["Processed Value: Hello world!"]
+
+  ;; build new producer
+  (do (def new-producer (build-producer))
+      new-producer)
+  #_=> #_#object[org.apache.kafka.clients.producer.KafkaProducer
+                 0x56863721
+                 "org.apache.kafka.clients.producer.KafkaProducer@56863721"]
+
+  ;; setup producer atom with new producer client
+  (setup-producer-atom-with-new-producer! new-producer)
+  #_=> #_#object[org.apache.kafka.clients.producer.KafkaProducer
+                 0x56863721
+                 "org.apache.kafka.clients.producer.KafkaProducer@56863721"]
+
+  ;; build new client
+  (do (def new-consumer (build-consumer))
+      new-consumer)
+  #_=> #_#object[org.apache.kafka.clients.consumer.KafkaConsumer
+                 0x1129a6bf
+                 "org.apache.kafka.clients.consumer.KafkaConsumer@1129a6bf"]
```

Comment:
This demonstration is likely unnecessary, as `build-producer` is demonstrated earlier, and the atom is also already demonstrated.